build_common.ps1

```diff
@@ -997,135 +775,221 @@ class BuildProject {
+						if ($outTxt.Contains("AkAudio") -or $outTxt.Contains("OnlineSubsystemSteamworks")) {
+							$event.MessageData.foundNativeScriptError = $true
+						} else {
+							$event.MessageData.foundRelevantZeroExitError = $true
+						}
+					}
+
+					CheckForCrash $outTxt $event.MessageData
+				}
+			}
+			default {
+				throw "unreachable"
+			}
+		}
 
-	# Explicitly set LASTEXITCODE from the process exit code so the rest of the script
-    # doesn't need to care if we launched the process in the background or via "&".
-    $global:LASTEXITCODE = $exitCode
+		# Create the process and register for the various events we care about.
+		$process = New-Object System.Diagnostics.Process
+		Register-ObjectEvent -InputObject $process -EventName OutputDataReceived -Action $outAction -MessageData $messageData | Out-Null
+		Register-ObjectEvent -InputObject $process -EventName ErrorDataReceived -Action $errAction | Out-Null
+		Register-ObjectEvent -InputObject $process -EventName Exited -Action $exitAction -MessageData $exitData | Out-Null
+		$process.StartInfo = $pinfo
+
+		# All systems go!
+		$process.Start() | Out-Null
+		$process.BeginOutputReadLine()
+		$process.BeginErrorReadLine()
+
+		# Wait for the process to exit. This is horrible, but using $process.WaitForExit() blocks
+		# the powershell thread so we get no output from make echoed to the screen until the process finishes.
+		# By polling we get regular output as it goes.
+		try {
+			while (!$exitData.exited) {
+				Start-Sleep -m 1
+			}	
+		}
+		finally {
+			# If we are stopping MSBuild hosted build, we need to kill the editor manually
+			if (!$exitData.exited) {
+				Write-Host "Killing $($cmdlet) tree"
+				KillProcessTree $process.Id
+			}
+		}
+
+		if ($messageData.crashDetected) {
+			ThrowFailure "Crash detected in $($cmdlet)"
+		}
+
+		$exitCode = $process.ExitCode
+		switch ($cmdlet)
+		{
+			"make" {
+				if ($exitCode -ne 0) {
+					ThrowFailure "Failed to compile scripts"
+				}
+			}
+			"precompileshaders" {
+				if ($exitCode -ne 0) {
+					ThrowFailure "Failed to compile shaders"
+				}
+			}
+			"cookpackages" {
+				if ($messageData.foundNativeScriptError) {
+					Write-Host ""
+					Write-Host "Detected errors about AkAudio and/or OnlineSubsystemSteamworks - these are safe to ignore."
+					Write-Host "If you want to get rid of them, you would need to build the mod in non-debug mode"
+					Write-Host "at least once - the errors will then go away (until the BuildCache folder is cleared/deleted)"
+					Write-Host ""
+				}
+
+				if ($messageData.foundRelevantZeroExitError) {
+					ThrowFailure "Found a relevant error while cooking (commandlet exited successfully but logs indicate error)"
+				}
+
+				# Backup in case our output parsing didn't catch something
+				if ((!$messageData.foundNativeScriptError) -and ($process.ExitCode -ne 0)) {
+					ThrowFailure "Cooker exited with non-0 exit code"
```

Comment:
If I'm reading this correctly, the `AkAudio`/`OnlineSubsystemSteamworks` errors cause a non-zero exit code even though they can be safely ignored?